runtime/common/src/lib.rs

```diff
@@ -121,37 +121,48 @@ pub mod types {
 
 	/// A representation of a tranche weight, used to weight
 	/// importance of a tranche
+	#[derive(Encode, Decode, Copy, Debug, Default, Clone, PartialEq, Eq, TypeInfo, CompactAs)]
+	#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+	pub struct TrancheWeight(pub u128);
+
+	/// A representation of InstanceId for Uniques.
 	#[derive(
-		codec::Encode,
-		codec::Decode,
-		Copy,
-		Debug,
+		Encode,
+		Decode,
 		Default,
+		Copy,
 		Clone,
 		PartialEq,
 		Eq,
+		CompactAs,
+		Debug,
+		MaxEncodedLen,
 		TypeInfo,
-		codec::CompactAs,
 	)]
 	#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-	pub struct TrancheWeight(pub u128);
+	pub struct InstanceId(pub u128);
 
-	/// A representation of InstanceId for Uniques.
+	/// A type describing the additional metadata stored in the OrmlAssetRegistry
 	#[derive(
-		codec::Encode,
-		codec::Decode,
-		Default,
-		Copy,
 		Clone,
+		Copy,
+		Default,
+		PartialOrd,
+		Ord,
 		PartialEq,
 		Eq,
-		codec::CompactAs,
 		Debug,
-		codec::MaxEncodedLen,
+		Encode,
+		Decode,
 		TypeInfo,
+		MaxEncodedLen,
 	)]
-	#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-	pub struct InstanceId(pub u128);
+	pub struct CustomMetadata {
+		pub fee_per_second: Option<Balance>,
+		pub mintable: bool,
+		pub permissioned: bool,
+		pub pool_currency: bool,
+	}
 }
```

Comment:
@offerijns appreciate getting your eyes on this. 

Is just a `permissioned: bool` enough for the `PreConditions` check? If I understand it correctly, we could check whether `pool_currency` is true and if so, check with the `permissions` pallet whether the right permissions are met.

Let me have your thoughts here.

Reply:
@branan

Reply:
On permissioned vs non-permissioned tokens, yes I believe that `permissioned: bool` could be checked using the new `Inspect` trait from the `RestrictedTokens` `PreConditions`, and thus would work perfectly to specify that a token is permissioned.

On the `pool_currency`, I'm not even 100% sure we need this, we could also just block tranche tokens as pool currencies in the existing `Contains` trait. But looks like you already implemented it and it might be useful at some point, so I think it makes sense 👍 

On implementing mintable permissions, we likely should add a `MintableCurrencyIssuer` role to the permissions pallet, instead of the current `PermissionedCurrencyIssuer`, as this can be used for both. The `mintable: bool` makes a lot of sense then.

Reply:
Cool, thanks @offerijns. Created an issue to capture the [`MintableCurrencyIssuer`](https://github.com/centrifuge/centrifuge-chain/issues/831) bit.